vuepress/docs/next/tutorials/create/ms/run-local.md

```diff
@@ -3,178 +3,157 @@ sidebarDepth: 2
 
-6.  Log in using:
+6. Log in using:
 
         Username: user
         Password: user
 
-7.  You’ll be redirected to the widget page and you can see the widget form with the ID 1 loaded.
-
-### Widget Details notes:
+7.  Once logged in, you can see the widget details with the ID 1 loaded
 
-If you want to load other data you have to change the index.html file in the public folder:
+#### Widget Details notes:
 
-    cd ui/widgets/YOUR-ENTITY-NAME/detailsWidget/public
-
-and change the "id" attribute in this line:
+If you want to use custom data, edit the `index.html` file in the public folder ui/widgets/YOUR-ENTITY-NAME/detailsWidget/public. Change the "id" attribute in this line:
 
     <my-entity-details service-url="%REACT_APP_SERVICE_URL%" id="1" />
 
-## Notes
-
-### Change Keycloak dev settings
 
-If you want to change your Keycloak settings to use another keycloak installation (not the docker-compose pre-configured one) or if you want to change the service-url of your widget you can change the parameters set in the `.env.local` file that was generated by the Entando Blueprint in the root folder of your react widgets:
+### Keycloak Settings and Issues 
+1. Change Keycloak Dev Settings
 
-    cd ui/widgets/YOUR-ENTITY-NAME/tableWidget
+If you want to use another Keycloak installation or change the service-URL of your widget, change the parameters set in the `.env.local` file that was generated by the Entando Blueprint in the root folder of your widgets. For instance, `ui/widgets/YOUR-ENTITY-NAME/tableWidget/.env.local`
 
-then edit the file `.env.local`
-
-By default this variables are set to:
+By default these variables are set to:
 
     REACT_APP_SERVICE_URL=http://localhost:8081/services/YOUR-APPLICATION-NAME/api
     REACT_APP_KEYCLOAK_URL=http://localhost:9080/auth
     REACT_APP_KEYCLOAK_REALM=jhipster
     REACT_APP_KEYCLOAK_CLIENT_ID=web_app
 
-### The service-url Variable
-
-The `service-url` variable is the Microservice API URL.
-
-### User is not authenticated message
-
-When you run the widgets if you see the message: `User is not authenticated`. This means that probably your keycloak application is not running so please check if the docker-compose command is still in execution.
+2. By default, Docker Compose will recreate the Keycloak container and reset the H2 database each time it is started. To persist Keycloak changes across restarts, you can make the following changes: 
+   * In your src/main/docker/keycloak.yml file, replace\
+   `
+   '-Dkeycloak.migration.strategy=OVERWRITE_EXISTING',` with 
+   ```
+   '-Dkeycloak.migration.strategy=IGNORE_EXISTING',
+   ```   
+   * In the same file, add a persistent volume under `volumes`:
+   ```
+   - ./keycloak-db:/opt/jboss/keycloak/standalone/data
+   ```   
+   * Keycloak changes should now be persistent. You can reset your Keycloak database by emptying the `src/main/docker/keycloak-db` directory and restarting the container.
```

Comment:
sometimes file paths are backticked and sometimes they're not; since all usage is the same we should be consistent